### PR TITLE
Structural/move update sensitivity call

### PR DESCRIPTION
**Pull request by MFusseder**
Within this PR I move  `UpdateSensitivities` from `SolveSolutionStep` to `FinalizeSolutionStep` of the adjoint structural python solver. Therefor I had to modifiy `structural_response.py` and `CalculateValue`of `AdjointNodalDisplacementResponseFunction`.

### CODE REVIEW

**Review by armingeiser — DISMISSED** [review]
Fine for me, did you verify with @dbaumgaertner?

File: applications/StructuralMechanicsApplication/python_scripts/structural_response.py [hunk 1]
```diff
@@ -307,57 +306,39 @@ def InitializeSolutionStep(self):
...
     def CalculateGradient(self):
-        Logger.PrintInfo("\n> Starting adjoint analysis for response:", self.identifier)
+        # synchronize the modelparts
+        self._SynchronizeAdjointFromPrimal()
         startTime = timer.time()
-        self.adjoint_analysis._GetSolver().Predict()
-        self.adjoint_analysis._GetSolver().SolveSolutionStep()
+        Logger.PrintInfo("\n> Starting adjoint analysis for response:", self.identifier)
+        if not self.adjoint_analysis.time < self.adjoint_analysis.end_time:
+            self.adjoint_analysis.end_time += 1
```
Comment (philbucher) [inline applications/StructuralMechanicsApplication/python_scripts/structural_response.py:322]: hm this looks like quite a specific hack
Why is it needed?

File: applications/StructuralMechanicsApplication/custom_response_functions/response_utilities/adjoint_nodal_displacement_response_function.cpp [hunk 2]
```diff
@@ -182,7 +182,9 @@ namespace Kratos
         const VariableComponentType& r_traced_dof =
             KratosComponents<VariableComponentType>::Get(mTracedDofLabel);
 
-        return mpTracedNode->FastGetSolutionStepValue(r_traced_dof, 0);
+        const auto& r_primal_traced_node = rModelPart.GetNode(mpTracedNode->Id());
+
+        return r_primal_traced_node.FastGetSolutionStepValue(r_traced_dof, 0);
```
Comment (philbucher) [inline applications/StructuralMechanicsApplication/custom_response_functions/response_utilities/adjoint_nodal_displacement_response_function.cpp]:
```suggestion
        return rModelPart.GetNode(mpTracedNode->Id()).FastGetSolutionStepValue(r_traced_dof, 0);
```
then you don't need the temp-var
A little less clean though, therefore just a suggestion :)

File: applications/StructuralMechanicsApplication/python_scripts/structural_response.py (hunk 1 above)
Comment (MFusseder) [reply applications/StructuralMechanicsApplication/python_scripts/structural_response.py:322]: To be honest I just copied it from the primal problem above. I can remove it. @armingeiser is there a reason why you do this time check for the primal analysis?

Comment (armingeiser) [reply applications/StructuralMechanicsApplication/python_scripts/structural_response.py:322]: It is, but as Martin said, it is already done like this for the primal analysis. It is not related to this change.

**Review by armingeiser — APPROVED** [review]

**Pull request merged**